core/src/main/java/org/testcontainers/containers/Container.java

```diff
@@ -141,6 +141,18 @@ default void addFileSystemBind(final String hostPath, final String containerPath
      */
     SELF waitingFor(@NonNull WaitStrategy waitStrategy);
 
+    /**
+     * Configure the container using the provided consumer.
+     *
+     * @param consumer the consumer of the container
+     * @return this
+     */
+    default SELF with(Consumer<SELF> consumer)
```

Comment:
We're working on a new API that will eliminate the need for a fluent-friendly way of applying "traits".
Because of that, could you please move it to `GenericContainer`?

Reply:
Done